Add article on binding to Kotlin

jjatria opened the pull request:
So delayed, but at least I had a lot of fun writing it. Thanks for the patience, and the prodding! I look forward to any feedback you might have.

Also: is there a way to get highlighting for the Kotlin and C bits?

oalders commented:
> Also: is there a way to get highlighting for the Kotlin and C bits?

If `vim` can do it, then we can as well. You can grep existing articles for `!#vim` and see where we've added highlighting for other languages.

If you want to update with this syntax, please feel free and I'll check how it renders before merging. You can preview locally, but right now it involves a few steps that I don't want to put you through. Maybe a project for next Christmas.

Also, you can move the article to `./2023/articles/2023-12-24.pod` if you like, since that's the last open spot anyway. 😄

oalders commented on 2023/incoming/santa-kotlin-is-coming-to-perl.pod:
```diff
@@ -0,0 +1,275 @@
+I've always loved Perl. It was not my very first programming language (you and
+me, BASIC, for life), but it was the first one where I felt like I could write
+real programs. The first that I felt was worth mastering, and the one I'm most
+comfortable, even today.
```
```suggestion
comfortable with, even today.
```

oalders commented on 2023/incoming/santa-kotlin-is-coming-to-perl.pod:
```diff
@@ -0,0 +1,275 @@
+
+Lucky for me, they had made the code of those SDKs publicly available, so I
+could examine it. And while looking through them I realised that most of the
+heavy-lifting was done by binding to a shared C library. My teenager-self
```
```suggestion
heavy lifting was done by binding to a shared C library. My teenager-self
```

oalders commented on 2023/incoming/santa-kotlin-is-coming-to-perl.pod:
```diff
@@ -0,0 +1,275 @@
+type C<Root>, its value will automatically be cast into a Santa::Kotlin::Root
+object.
+
+We still need to get our hands in an instance of this outermost struct, and for
```
```suggestion
We still need to get our hands on an instance of this outermost struct, and for
```

oalders commented on 2023/incoming/santa-kotlin-is-coming-to-perl.pod:
```diff
@@ -0,0 +1,275 @@
+if you want to skip all of that you can jump directly to the
+L<meat and potatoes|/Binding to Kotlin from Perl>.
+
+=head2 Stay awhile and listen
```
```suggestion
=head2 Stay a while and listen
```

jjatria replied:
Ah, that was my first thought as well! But then I had second thoughts, because the only reason I know of this line is because it's one of the lines that [Deckard Cain](https://diablo.fandom.com/wiki/Deckard_Cain) says in the original Diablo game, and it looks like in that context it is [consistently styled](https://diablo.fandom.com/wiki/Stay_Awhile_and_Listen) like "stay _awhile_ and listen" (which I also find odd, but...).

Do you reckon I should still change it?

oalders replied:
> Do you reckon I should still change it?

No. I didn't realize it was a reference. That makes sense. Also, my inclination is not to edit articles too heavily. I don't want to redact all of the personality and charm.

oalders reviewed (changes requested):
Great stuff! Just a few suggestions to consider.

jjatria commented:
I'm glad you liked it!

I've made all the corrections you suggested (except for the "awhile" -> "a while" one for now), and fixed the highlighting for Kotlin and C code. I've also moved the file over to the slot for December 24th.

oalders reviewed (approved):
\o/

oalders commented:
Highlighting looks good!

<img width="793" alt="Bildschirmfoto 2023-12-22 um 9 36 46 PM" src="https://github.com/perladvent/Perl-Advent/assets/96205/d604298c-e57c-4e03-8563-01e1b5a1431f">
